extension/src/stats_agg.rs

```diff
@@ -1884,44 +2187,92 @@ mod tests {
                 &pg_moment_samp_query(3, "test_x"),
                 &tk1d_agg_arg("skewness", "sample"),
                 BILLIONTH,
+                false,
             );
             check_agg_equivalence(
                 state,
                 &client,
                 &pg_moment_samp_query(3, "test_x"),
                 &tk2d_agg_arg("skewness_x", "sample"),
                 BILLIONTH,
+                false,
             );
             check_agg_equivalence(
                 state,
                 &client,
                 &pg_moment_samp_query(3, "test_y"),
                 &tk2d_agg_arg("skewness_y", "sample"),
                 BILLIONTH,
+                false,
             );
             check_agg_equivalence(
                 state,
                 &client,
                 &pg_moment_samp_query(4, "test_x"),
                 &tk1d_agg_arg("kurtosis", "sample"),
                 BILLIONTH,
+                false,
             );
             check_agg_equivalence(
                 state,
                 &client,
                 &pg_moment_samp_query(4, "test_x"),
                 &tk2d_agg_arg("kurtosis_x", "sample"),
                 BILLIONTH,
+                false,
             );
             check_agg_equivalence(
                 state,
                 &client,
                 &pg_moment_samp_query(4, "test_y"),
                 &tk2d_agg_arg("kurtosis_y", "sample"),
                 BILLIONTH,
+                false,
             );
 
             client.select("DROP TABLE test_table", None, None);
         });
     }
+
+    #[pg_test]
+    fn stats_agg_rolling() {
+        Spi::execute(|client| {
+            client.select(
+                "
+SET timezone TO 'UTC';
+CREATE TABLE prices(ts TIMESTAMPTZ, price FLOAT);
+INSERT INTO prices (
+    WITH dates AS
+        (SELECT
+            *
+        FROM
+            generate_series('2020-01-01 00:00'::timestamp, '2020-02-01 12:00', '10 minutes') time)
+    SELECT
+        dates.time,
+        (select (random()+EXTRACT(seconds FROM dates.time))*100 ) price
+    FROM
+        dates
+);
+",
+                None,
+                None,
+            );
+
+            let mut vals = client.select(
+                "SELECT stddev(data.stats_agg) FROM (SELECT stats_agg(price) OVER (ORDER BY ts RANGE '50 minutes' PRECEDING) FROM prices) data",
+                None, None
+            );
+            assert!(vals.next().unwrap()[1].value::<f64>().unwrap().is_nan());
+            assert!(vals.next().unwrap()[1].value::<f64>().is_some());
+            assert!(vals.next().unwrap()[1].value::<f64>().is_some());
+
+            let mut vals = client.select(
+                "SELECT slope(data.stats_agg) FROM (SELECT stats_agg((EXTRACT(minutes FROM ts)), price) OVER (ORDER BY ts RANGE '50 minutes' PRECEDING) FROM prices) data;",
+                None, None
+            );
+            assert!(vals.next().unwrap()[1].value::<f64>().is_none()); // not sure why this is None?
```

Comment:
There's only one point in that one so slope's undefined I believe.

Reply:
Right, documented.